core/rawdb/accessors_batch_event.go

```diff
@@ -0,0 +1,157 @@
+package rawdb
+
+import (
+	"bytes"
+	"math/big"
+	"strconv"
+
+	"github.com/scroll-tech/go-ethereum/common"
+	"github.com/scroll-tech/go-ethereum/ethdb"
+	"github.com/scroll-tech/go-ethereum/log"
+	"github.com/scroll-tech/go-ethereum/rlp"
+)
+
+// ChunkRange represents the range of blocks within a chunk.
+type ChunkRange struct {
+	StartBlockNumber uint64
+	EndBlockNumber   uint64
+}
+
+// FinalizedBatchMeta holds metadata for finalized batches.
+type FinalizedBatchMeta struct {
+	BatchHash            common.Hash
+	TotalL1MessagePopped uint64
```

Comment:
Is `TotalL1MessagePopped` before, after, or in this batch? Let's add a comment to clarify.

Reply:
before + in this batch.

Reply:
a comment is added in [ba66df0](https://github.com/scroll-tech/go-ethereum/pull/515/commits/ba66df0ebcf2f7db9aa0b3203906c1f02cee623f)